common/src/legacy_memory_region.rs

```diff
@@ -42,18 +45,32 @@ where
     /// library assumes that references can never point to virtual address `0`.  
     pub fn new(memory_map: I) -> Self {
         // skip frame 0 because the rust core library does not see 0 as a valid address
-        let start_frame = PhysFrame::containing_address(PhysAddr::new(0x1000));
-        Self::new_starting_at(start_frame, memory_map)
+        // also skip the first 1MB of frames, there are use cases that require lower conventional memory access. (Such as SMP SIPI)
+        let start_frame = PhysFrame::containing_address(PhysAddr::new(LOWER_MEMORY_END_PAGE));
+        unsafe { Self::unsafe_new_starting_at(start_frame, memory_map) }
```

Comment:
With the suggested changes below, we could keep the previous implementation:

```suggestion
        let start_frame = PhysFrame::containing_address(PhysAddr::new(0x1000));

        Self::new_starting_at(start_frame, memory_map)
```